src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -272,24 +369,34 @@ CHIP_ERROR SilabsTracer::LoadWatermarks()
     return CHIP_NO_ERROR;
 }
 
-#if CONFIG_BUILD_FOR_HOST_UNIT_TEST
-size_t SilabsTracer::GetTraceCount()
-{
-    return mBufferedTrackerCount;
-}
-
-CHIP_ERROR SilabsTracer::GetTraceByIndex(size_t index, const char *& trace) const
+CHIP_ERROR SilabsTracer::GetTraceByOperation(TimeTraceOperation aOperation, char * buffer, size_t & size) const
 {
     auto * current = mTimeTrackerList.head;
-    for (size_t i = 0; i < index && current != nullptr; ++i)
+    while (current != nullptr)
     {
+        if (current->mValue.mOperation == aOperation)
+        {
+            // Check if the buffer is large enough
+            auto requiredSize = current->mValue.ToCharArray(nullptr, 0);
```

Comment:
Having guetter that returns the size would make more sense than `ToCharArray(nullptr, 0)` which seems ominous